File: Data Structures & Algorithms/valid-binary-search-tree/submission-6.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isValidBST(TreeNode root) {
        if(root==null) return true;
        return valid(root, Integer.MIN_VALUE, Integer.MAX_VALUE);
    }
    public boolean valid(TreeNode root, int mini, int maxm) {
        if(root==null) return true;
        if(root.val>=maxm || root.val<=mini)
            return false;
        return valid(root.left,mini,root.val) && valid(root.right, root.val,maxm);
    }
}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-6.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        if not root:
            return ans
        def traverse(node: Optional[TreeNode]):
            if not node:
                return
            traverse(node.left)
            ans.append(node.val)
            traverse(node.right)
        traverse(root)
        return ans